app/src/main/java/com/quran/labs/androidquran/widgets/HighlightingImageView.java

```diff
@@ -130,17 +152,178 @@ public void setNightMode(boolean isNightMode, int textBrightness) {
     adjustNightMode();
   }
 
-  public void highlightAyah(int sura, int ayah, HighlightType type) {
+  private void highlightFloatableAyah(Set<String> highlights, String currentSurahAyah) {
+    final Map<String, List<AyahBounds>> coordinatesData = ayahCoordinates.getAyahCoordinates();
+
+    String previousSurahAyah = currentSurahAyah;
+    for(String surahAyahs: highlights) {
+      previousSurahAyah = surahAyahs;
+    }
```

Comment:
this loop will basically just get the "last" one, and that perhaps can change - what are you actually trying to get here then?

Reply:
Right. So this is me being lazy. All I want to do here is simply get the only element from the Set (I know this because, I have checked the size previously). So instead of doing it the obvious iterator way, I followed this approach.

I have fixed this.